### ISSUE TITLE
Machine learning algorithms and data visualisation for airfoil sound pressure level prediction

### ISSUE THREAD

**Issue by amukh18** [WebClub-NITK/Hacktoberfest-2k19#291]
### Description

Your task is to predict the scaled sound pressure level(dB) from aerodynamic and acoustic tests of two and three-dimensional airfoil blade section conducted in an anechoic wind tunnel. 
This is a **regression** problem.
An airfoil is a shaped surface such as an airplane wind, tail, or propeller blade that produces lift and drag when moved through the air.
An anechoic wind tunnel is a type of wind tunnel that is designed to completely absorb sound waves.

The dataset can be found [here](https://archive.ics.uci.edu/ml/datasets/Airfoil+Self-Noise).

The features convey the following information:
  **1.** Frequency, in hertz.
  **2.** Angle of attack(in degrees).
  **3.** Chord length, in meters.
  **4.** Free-stream velocity, in metres per second.
  **5.** Suction side displacement thickness, in meters.
  **6.** Scaled sound pressure level, in dB.

### Details
- Technical Specifications: pandas, python, sci-kit, numpy
- Type of issue: Multiple
- Time Limit: 
  - Plots must be implemented within **24 hours** of being taken up.
  - Algorithms must be implemented within **2 days** of being taken up.
  - Cross-validations/ensembles must be implemented within **3 days** of being taken up.
  
**The folder associated with the issue contains an IPython notebook containing a train-test split to get you started. It also contains the data set in CSV format.**

### Issue requirements / progress
All algorithms and ensembles must be scores using *RMSE*, *Logloss* and *Accuracy* metrics. Each pull request must only fulfill one of the tasks below.

#### Plots:
- [ ] Box plot (2 vs 4)
- [ ] Heatmap
- [ ] Linear regression model

#### Algorithms:
- [ ] Support vector machine
- [ ] K-nearest neighbors
- [ ] Random forest regressor
- [ ] XGBoost 
- [ ] LightGBM 
- [ ] Multi-layer Perceptron 
- [ ] Decision Tree Regressor

#### Cross-validations/Ensembles:
- [ ] 10-fold cross-validation of XGBoost
- [ ] 10-fold cross-validation of LightGBM
- [ ] Average (non-weighted) the predictions of a 10-fold cross-validation of XGBoost and LightGBM (both must be implemented in the fold)

---

### Resources
List of resources that might be required / helpful.
Here are a few resources that may help you:
1. NumPy documentation: https://docs.scipy.org/doc/numpy-1.13.0/reference/index.html
2. Scikit-learn documentation: https://scikit-learn.org/stable/documentation.html
3. Pandas documentation: https://pandas.pydata.org/pandas-docs/stable/
4. Jupyter Notebook installation and tutorial :	https://www.dataquest.io/blog/jupyter-notebook-tutorial/
5. XGBoost documentation: https://xgboost.readthedocs.io/en/latest/
6. LightGBM documentation: https://lightgbm.readthedocs.io/en/latest/
7. Scikit-learn documentation
	- Support vector machine: hhttps://scikit-learn.org/stable/modules/generated/sklearn.svm.SVR.html
	- K-nearest neighbors: https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.KNeighborsRegressor.html
    - Gaussian Naive Bayes: https://scikit-learn.org/stable/modules/generated/sklearn.naive_bayes.GaussianNB.html
	- Random forest regressor : https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.RandomForestRegressor.html
	- Multi-layer perceptron regressor: https://scikit-learn.org/stable/modules/generated/sklearn.neural_network.MLPRegressor.html
	- Decision tree regressor: https://scikit-learn.org/stable/modules/generated/sklearn.tree.DecisionTreeRegressor.html
8.	Seaborn documentation
	a. Box plot: https://seaborn.pydata.org/generated/seaborn.boxplot.html
	b. Heat-map: http://seaborn.pydata.org/generated/seaborn.heatmap.html
	c. Linear regression model:https://seaborn.pydata.org/generated/seaborn.regplot.html
---

### Directory Structure
The following convention must be adhered to when placing your solution files.
#### Plots:
- For Box-plot
`/machineLearning/airfoil/plots/bp/<solution_file>`
- For Heatmap
`/machineLearning/airfoil/plots/hm/<solution_file>`
- For Linear Regression Model
`/machineLearning/airfoil/plots/lrm/<solution_file>`

#### Algorithms:
-   For Support Vector Machine:  
    `/machineLearning/airfoil/algo/svm/<solution_file>`
-	For K-Nearest Neighbors: 
	`/machineLearning/airfoil/algo/knn/<solution_file>`
-	For Decision Trees Regressor:
	`/machineLearning/airfoil/algo/dtm/<solution_file>`
-	For Random Forest Regressor:
	`/machineLearning/airfoil/algo/rfm/<solution_file>`
- 	For Multi-layer Perceptron:
	`/machineLearning/airfoil/algo/mlp/<solution_file>`
-	For XGBoost:
	`/machineLearning/airfoil/algo/xgb/<solution_file>`
-	For LightGBM:
	`/machineLearning/airfoil/algo/lgbm/<solution_file>`

#### Ensembles:
-   For 10-fold XGBoost:  
    `/machineLearning/airfoil/ens/10-xgb/<solution_file>`
- 	For 10-fold LightGBM:
	`/machineLearning/airfoil/ens/10-lgbm/<solution_file>`
- 	For average(non-weighted) of predictions of 10-fold XGBoost and 10-fold LightGBM
	`/machineLearning/airfoil/ens/avg/<solution_file>`

#### Note
Please claim the issue first by commenting here before starting to work on it. Feel free to contact @amukh18 or @CinnamonRolls1 with any issues at any time.
